### PR TITLE
Collect Artifact Stats

### PR DESCRIPTION
**Pull request by bialesdaniel**
This creates the test runner that will collect artifact stats. It is designed to be very easy to add new "collectors".

Other changes:
- I decoupled the DB server code from the TestServer class. This allowed me to create instances of the DB in the artifact stats collectors. 
- A bunch of small cosmetic changes when I ran autopep8

A lot of files changed cause of the autopep8 thing. I also merged in the changes from collect_mem_info so those changes will appear here. The idea is that will get merged in before this. These are the files to pay attention to:
- anything under `artifact_stats/`
- `test_server.py`
- `db_server.py`
- `jenkinsfile-nightly`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -275,6 +275,7 @@ wal.log
 
 # OLTP output
 **/oltp_output_*
+**/oltp_result
```
Comment (jazzplato) [inline .gitignore]: is it a folder named as `oltp_bench`?

File: Dockerfile [hunk 2]
```diff
@@ -11,4 +11,5 @@ COPY script/installation/packages.sh install-script.sh
 RUN echo y | ./install-script.sh all
 
 COPY . /repo
-WORKDIR /repo/build
+#WORKDIR /repo/build
+WORKDIR /repo
```
Comment (jazzplato) [inline Dockerfile]: love this!!!

File: script/testing/util/test_server.py [hunk 3]
```diff
@@ -1,41 +1,42 @@
...
+        db_output_file = args.get("db_output_file", constants.DEFAULT_DB_OUTPUT_FILE)
+        db_host = args.get("db_host", constants.DEFAULT_DB_HOST)
+        db_port = args.get("db_port", constants.DEFAULT_DB_PORT)
+        build_type = args.get("build_type", "")
+        server_args = args.get("server_args", "").strip()
 
-        # db server location
-        self.db_host = self.args.get("db_host", constants.DEFAULT_DB_HOST)
-        self.db_port = self.args.get("db_port", constants.DEFAULT_DB_PORT)
+        self.db_instance = NoisePageServer(db_host, db_port, build_type, server_args, db_output_file)
```
Comment (jazzplato) [inline script/testing/util/test_server.py]: This is neat and elegant

**Review by jazzplato — COMMENTED** [review]
Looks good to me in general. Make sure to merge with the [PR-1243](https://github.com/cmu-db/noisepage/pull/1243) first because there have been a number of changes from the current version here.

File: .gitignore (hunk 1 above)
Comment (bialesdaniel) [reply .gitignore]: I honestly don't remember adding this. I think the OLTPBench tests create this directory but let me double-check that.

Comment (bialesdaniel) [reply .gitignore]: yup its a file generated when the OLTPBench tests run

**Pull request closed without merging**